TKeazirian.HTTPServer/Server.cs

```diff
@@ -23,16 +23,16 @@ public static void StartListening()
 
             while (true)
             {
+                Router router = new Router();
                 var socket = listener.Accept();
 
                 _request = GetRequest(socket);
 
-                string response = Controller.EchoRequestBody(_request);
-
+                var response = router.HandleRequest(_request);
                 byte[] encodedResponse = Encoding.ASCII.GetBytes(response);
 
                 socket.Send(encodedResponse, SocketFlags.None);
-                socket.Close();
+                SocketHandler.CloseSocketConnection(socket);
```

Comment:
I like these names, they give a good sense of what is going on without having to drill down into the details.